jwt.js

```diff
@@ -1,61 +1,114 @@
-'use strict'
-
 const fp = require('fastify-plugin')
 const JWT = require('jsonwebtoken')
-const assert = require('assert')
+const steed = require('steed')()
+
+function wrapStaticSecretInCallback (secret) {
+  return function (_, __, cb) {
+    return cb(null, secret)
+  }
+}
 
-function fastifyJWT (fastify, opts, next) {
-  if (!opts.secret) {
+function fastifyJwt (fastify, options, next) {
+  if (!options.secret) {
     return next(new Error('missing secret'))
   }
 
-  const secret = opts.secret
+  let secretCallback = options.secret
+
+  if (typeof secretCallback !== 'function') { secretCallback = wrapStaticSecretInCallback(secretCallback) }
+
+  let _requestProperty = options.userProperty || options.requestProperty || 'user'
+  let _resultProperty = options.resultProperty
+  const credentialsRequired = typeof options.credentialsRequired === 'undefined' ? true : options.credentialsRequired
```

Comment:
Why was this option introduced? Could you remove it?

Reply:
This is another thing I abstracted from `express-jwt`. After reviewing `node-jsonwebtoken` it seems like this is a not an actual option. I'll remove it and update api.